arangod/Pregel/Algos/WCC.cpp

```diff
@@ -96,6 +102,16 @@ struct WCCComputation
         }
       }
     }
+
+    if (globalSuperstep() == 1) {
+      auto range = this->allocateAuxiliaryStorage(inNeighborsByteSize);
+      myData.inNeighbors.init(range, inNeighbors.size());
+
+      for (auto const& inNeighbor : inNeighbors) {
+        myData.inNeighbors.emplace(inNeighbor);
+      }
+      inNeighbors.clear();
```

Comment:
What is the purpose of clearing `inNeighbors` here?
`clear()` doesn't free the underlying memory, and it is only done conditionally (superstep == 1). Plus, `inNeighbors` is a local variable, which would go out of scope at the end of the function (next line) anyway. I think there is no benefit of clearing the vector here, unless I am missing something. (that side, clearing here also does no harm).

Reply:
I moved the collection of `inNeighbors` to a separate function; as a bonus we save the dance of allocating a local variable on *every* computation to not use it most of the time.